Warsaw_Tram/MapViewController.swift

```diff
@@ -10,50 +10,147 @@ import UIKit
 import MapKit
 import CoreLocation
 
-class MapViewController: UIViewController, MKMapViewDelegate, CLLocationManagerDelegate {
+class MapViewController: UIViewController {
 
     @IBOutlet var map: MKMapView!
     
     var locationManager = CLLocationManager()
+    var alertHelper: AlertHelper?
+    var userLocationDefined: Bool = false
+    var userLocation: CLLocationCoordinate2D?
+    
+    let resourceId = "id=c7238cfe-8b1f-4c38-bb4a-de386db7e776"
+    let warsawTramsApiKey = "apikey=060b903c-b0c3-427e-934d-9e4a81a61969"
+    let mapLatDelta: CLLocationDegrees = 0.05
+    let mapLonDelta: CLLocationDegrees = 0.05
     
     override func viewDidLoad() {
-        
+        alertHelper = AlertHelper()
+        setupMap()
+        setupLocationManager()
+        super.viewDidLoad()
+    }
+    
+    @IBAction func showCurrentLocation(sender: AnyObject) {
+        showUserCurrentLocation()
+    }
+    
+    private func setupLocationManager() {
         locationManager.delegate = self
         locationManager.desiredAccuracy = kCLLocationAccuracyBest
         locationManager.requestWhenInUseAuthorization()
         locationManager.startUpdatingLocation()
-        
-        super.viewDidLoad()
-        addHomeAnnotation()
     }
     
-    func locationManager(manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        let userLocation: CLLocation = locations[0] as CLLocation
-        let latitude = userLocation.coordinate.latitude
-        let longitude = userLocation.coordinate.longitude
-        let latDelta: CLLocationDegrees = 0.01
-        let lonDelta: CLLocationDegrees = 0.01
-        let span: MKCoordinateSpan = MKCoordinateSpanMake(latDelta, lonDelta)
-        let location: CLLocationCoordinate2D = CLLocationCoordinate2DMake(latitude, longitude)
-        let region: MKCoordinateRegion = MKCoordinateRegionMake(location, span)
-        
-        self.map.setRegion(region, animated: true)
-    }
-    
-    private func addHomeAnnotation() {
-        let latitude: CLLocationDegrees = 52.453946
-        let longitude: CLLocationDegrees = 19.005797
-        let latDelta: CLLocationDegrees = 0.05
-        let lonDelta: CLLocationDegrees = 0.05
-        let span: MKCoordinateSpan = MKCoordinateSpanMake(latDelta, lonDelta)
-        let location:CLLocationCoordinate2D = CLLocationCoordinate2DMake(latitude, longitude)
-        let region:MKCoordinateRegion = MKCoordinateRegionMake(location, span)
-        map.setRegion(region, animated: true)
+    private func setupMap() {
+        navigationController?.navigationBar.backgroundColor = UIColor.orangeColor()
+        map.showsScale = true
+    }
+    
+    private func fetchTramsData() {
+        let stringURL : NSString = "https://api.um.warszawa.pl/api/action/wsstore_get?\(resourceId)&\(warsawTramsApiKey)"
```

Comment:
When I had used it at the beginning the NSURL returned me nil so I found this solution on stackoverflow. But now it works with String as well.